Providers/CoreServiceProvider.php

```diff
@@ -244,4 +247,50 @@ private function getCentralisedTranslationPath(Module $module)
     {
         return $this->app['modules']->find('Translation')->getPath() . "/Resources/lang/{$module->getLowerName()}";
     }
+
+    /**
+     * List of Custom Blade Directives
+     */
+    public function bladeDirectives()
+    {
+        /**
+         * Set variable.
+         * Usage: @set($variable, value)
+         */
+        Blade::directive('set', function ($expression) {
+            list($variable, $value) = $this->getArguments($expression);
+
+            return "<?php {$variable} = {$value}; ?>";
+        });
+
+        /**
+         * Php explode()
+         * Usage: @explode($delimiter, $string)
+         */
+        Blade::directive('explode', function ($expression) {
+            list($delimiter, $string) = $this->getArguments($expression);
+
+            return "<?php echo explode({$delimiter}, {$string}); ?>";
+        });
+
+        /**
+         * php implode()
+         * Usage: @implode($delimiter, $array)
+         */
+        Blade::directive('implode', function ($expression) {
+            list($delimiter, $array) = $this->getArguments($expression);
+
+            return "<?php echo implode({$delimiter}, {$array}); ?>";
+        });
+    }
+
+    /**
+     * Get argument array from argument string.
+     * @param $argumentString
+     * @return array
+     */
+    private function getArguments($argumentString)
+    {
+        return explode(', ', str_replace(['(', ')'], '', $argumentString));
```

Comment:
You need to be aware that in Laravel 5.3, Blade directives don't send along the outer parenthesis anymore.

Also, you could use [`str_getcsv`](http://php.net/manual/en/function.str-getcsv.php) to split the arguments:

```php
return str_getcsv(mb_substr($argumentString, 1, mb_strlen($argumentString) - 2), ',', "'"); // mb_substr removes the beginning and ending parenthesis. Once we move to Laravel 5.3, this wont be neccessary; we could just pass $argumentString.
```

Or something of that nature.